devicetypes/smartthings/aeotec-child-siren.src/aeotec-child-siren.groovy

```diff
@@ -0,0 +1,81 @@
+/**
+ *	Copyright 2019 SmartThings
+ *
+ *	Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *	in compliance with the License. You may obtain a copy of the License at:
+ *
+ *		http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *	Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *	on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *	for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+    definition(name: "Aeotec Child Siren", namespace: "smartthings", author: "SmartThings", mnmn: "SmartThings", vid: "SmartThings-smartthings-Z-Wave_Siren", ocfDeviceType: "x.com.st.d.siren") {
+        capability "Actuator"
+        capability "Health Check"
+        capability "Alarm"
+
+        command "reset"
+    }
+    tiles {
+        standardTile("alarm", "device.alarm", width: 2, height: 2) {
+            state "off", label: 'off', action: 'alarm.strobe', icon: "st.alarm.alarm.alarm", backgroundColor: "#ffffff"
+            state "both", label: 'alarm!', action: 'alarm.off', icon: "st.alarm.alarm.alarm", backgroundColor: "#e86d13"
+        }
+        standardTile("off", "device.alarm", inactiveLabel: false, decoration: "flat") {
+            state "default", label: '', action: "alarm.off", icon: "st.secondary.off"
+        }
+
+        main "alarm"
+        details(["alarm", "off"])
+    }
+}
+
+def installed() {
+    configure()
+    sendEvent(name: "alarm", value: "off", isStateChange: true)
+}
+
+def updated() {
+    configure()
+}
+
+def configure() {
+    sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false)
+}
+
+def off() {
+    parent.setOffChild(device.deviceNetworkId, 0x00)
+    parent.resetActiveSounds()
+
+}
+
+def on() {
+    parent.setOnChild(device.deviceNetworkId, 0xFF)
+    parent.setActiveSound(device.endpointId)
```

Comment:
why use `device.deviceNetworkId` in one call and `device.endpointId` in the next, if you're immediately pulling the endpointId out of the DNI? It's much simpler to re-construct the DNI from the endpointId than to extract the endpointId from the DNI.

If `device.endpointId` is actually defined, you should be using that.